### PR TITLE
initial loggging module

### PR DESCRIPTION
**Pull request by DiannaBanana**
Hey guys, here is an initial logging module. It works (you can see in logfile1.xml) but there is a delay from 1 user and it logs some things out of order. From the architectures we saw today, I think it would be a good idea to optimize it to something like an audit service so its more efficient & overall speedier. Check it out and lmk!

### CODE REVIEW

File: backend/src/user_commands.py [hunk 1]
```diff
@@ -20,45 +29,100 @@ def printCmd(cmdDict):
...
+    buyAmtQ.put(cmdDict['amount'])
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['userCommand'](cmdDict)
 
 def CMD_CommitBuy(cmdDict):
     printCmd(cmdDict)
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['userCommand'](cmdDict)
+    # The users account is modified after committing to buying the stock, log this action
+    cmdDict['timestamp'] = str(int(time.time()))
```
Comment (DamonAknuh) [inline backend/src/user_commands.py:59]: duplicate line on 56.

File: backend/src/user_commands.py [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+from audit import logXML as log
 from threading import Lock
 cmd_print_lock = Lock()
 
+buyAmtQ = queue.Queue()
+sellAmtQ = queue.Queue()
+REMOVE = "REMOVE"
+ADD = "ADD"
+
 def printCmd(cmdDict):
```
Comment (DamonAknuh) [inline backend/src/user_commands.py:23]: not a big deal, but It looks like all the user commands use both of the following: 

```
cmdDict['timestamp'] = str(int(time.time()))
and 
log.logEvents['accountTransaction' || 'quoteServer' ](cmdDict) || 
```

could we generalize it in this function?

Comment (DamonAknuh) [reply backend/src/user_commands.py:23]: oh, maybe not because they have specific error messages?

File: backend/src/user_commands.py (hunk 1 above)
Comment (DiannaBanana) [reply backend/src/user_commands.py:59]: There are 2 dfferent timestamps for the different events (first userCommand when the user issues the command then accountTransaction when the buy is committed). I wasn't sure how long the logic between them would take so I figured timestamp it twice?

File: backend/src/web_server.py [hunk 3]
```diff
@@ -66,6 +66,7 @@ def shutdown(self):
         except socket.error as err:
             pass
 
+# Change to create thread per user (currently recieves commands out of order)
```
Comment (DamonAknuh) [inline backend/src/web_server.py:69]: Good idea! I'll work on that! 

that would probably work a lot better with scaling?

File: backend/src/user_commands.py (hunk 2 above)
Comment (DiannaBanana) [reply backend/src/user_commands.py:23]: yea it defs need some optimizing, mostly I think bc we havent implemented the commands yet so its a bit unclear which/what actions need to be logged

File: backend/src/user_commands.py (hunk 1 above)
Comment (DamonAknuh) [reply backend/src/user_commands.py:59]: I see! yep fair enough!

**Review by DamonAknuh — APPROVED** [review]

**Review by skosman — APPROVED** [review]

**Review by skosman — APPROVED** [review]

**Pull request merged**